blog-posts/sls-security-headers/sls-security-headers.md

```diff
@@ -0,0 +1,96 @@
+
+HTTP security headers are a mechanism to **protect your application from common attacks**. HTTP security headers are, as their name suggests, a set of HTTP headers with security purposes.
+
+The table below shows a subset of HTTP security headers. The value is strongly recommended. The description mentions the common attacks that the header protects against.
+
+| Name | Value | Description |
+| --- | --- | --- |
+| [Strict-Transport-Security](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security) | max-age=31536000 | Informs browsers that the site should only be accessed using HTTPS, and that any future attempts to access it using HTTP should automatically be converted to HTTPS.
+The user's inputs will never be readable by a man on the middle (except potentially the first call ever, which does not contain any sensitive information). |
+| [X-Content-Type-Options](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Content-Type-Options) | nosniff | Prevents older browsers from sniffing the content of files sent by the server rather than trusting the Content-Type header. This prevents the detection and execution of malicious Javascript in an image uploaded by an attacker and then retrieved by a victim. |
+| [X-Frame-Options](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options) | SAMEORIGIN | Prevents the display of the site in an iframe, and thus protects against click-jacking. |
+| [X-XSS-Protection](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-XSS-Protection) | 1; mode=block | Prevents rendering of the page if an XSS attack is detected. |
+| [Referrer-Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Referrer-Policy) | strict-origin-when-cross-origin | Prevents sending sensitive information (origin, path and query string) to another website or when the protocol is downgraded (HTTPS → HTTP) |
+
+An extensive list can be found on the Mozilla Developer Network (MDN).
+
+[https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers#security](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers#security)
+
+HTTP security headers are not *per se* good coding practices, but rather good running strategies. As a result, they fall under the responsibility of ops teams.
+
+Let’s digress a little bit to mention the Cross Origin Resource Sharing (CORS) headers. As a web application developer, you certainly already have come across them. Otherwise, believe me, you will encounter them soon or later. 😉
+
+CORS headers are different because:
+
+- weaken a security feature of the browser of your users, [reformulate]
+- are a pain to deal with in your development journey. 😅
+
+These headers do not impact the application itself. Thus adding these can be done at a very low cost.
+
+
+# Demystifying HTTP security headers on AWS
+
+AWS mentions [several methods](https://aws.amazon.com/websites/) to deploy your website.
```

Comment:
Add a sentence after this one to explain that you are going through the two methods to deploy a website, and explain if it is possible and how to configure the security headers in each one.